frontend/app/src/store/history/mutations.ts

```diff
@@ -27,13 +27,64 @@ export const mutations: MutationTree<HistoryState> = {
     state.trades = trades;
   },
 
+  [HistoryMutations.ADD_TRADE_LOCATION_TO_PURGE](
+    state: HistoryState,
+    location: TradeLocation
+  ) {
+    state.tradeLocationsToPurge = [...state.tradeLocationsToPurge, location];
+  },
+
+  [HistoryMutations.REMOVE_TRADE_LOCATION_TO_PURGE](
+    state: HistoryState,
+    location?: TradeLocation
+  ) {
+    if (!location) {
+      state.tradeLocationsToPurge = [];
+      return;
+    }
+
+    const locations = [...state.tradeLocationsToPurge];
+    const locationIndex = locations.indexOf(location);
+    if (locationIndex > -1) {
+      locations.splice(locationIndex, 1);
+      state.tradeLocationsToPurge = [...locations];
+    }
+  },
+
```

Comment:
Wouldn't it be simpler to just trigger an `onlyCache: false` trade fetch when adding an exchange?

Reply:
I thought about that too. But I thought of another scenario. 
If I just trigger `onlyCache: false` for a single exchange when adding an exchange, and then the user open the trade page, which will trigger `onlyCache: true` only, if the `onlyCache: false` before still not loaded, then when the user the open trade page, the data is still not updated. I decided to put it on a state, so if user open the trade page and there are `onlyCache: true` query that aren't done yet, just call it again.

Reply:
But maybe it's too much. If you find it unnecessary, I will remove it

Reply:
Let's remove it, we will re-evaluate the logic there either way

Reply:
updated